# Análise exploratória de dados

In [1]:
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt

## 1. Descrição dos dados

### 1.1 Descrição do conjunto de dados

O conjunto de dados "Titanic - Machine Learning From Disaster" contém informações detalhadas sobre as vítimas do Titanic, que naufragou em 1912. A base de dados possui 891 entradas e 15 colunas, representando diversas características dos passageiros, incluindo dados demográficos, informações sobre a viagem e a sobrevivência. Algumas colunas têm valores ausentes, como idade e convés. Este dataset permite explorar fatores que influenciaram a sobrevivência no naufrágio.

### 1.2 Conjunto de dados

In [2]:
df = pd.read_csv("../data/raw/data.csv")
df

survived  pclass     sex   age  sibsp  parch     fare embarked   class  \
0           0       3    male  22.0      1      0   7.2500        S   Third   
1           1       1  female  38.0      1      0  71.2833        C   First   
2           1       3  female  26.0      0      0   7.9250        S   Third   
3           1       1  female  35.0      1      0  53.1000        S   First   
4           0       3    male  35.0      0      0   8.0500        S   Third   
..        ...     ...     ...   ...    ...    ...      ...      ...     ...   
886         0       2    male  27.0      0      0  13.0000        S  Second   
887         1       1  female  19.0      0      0  30.0000        S   First   
888         0       3  female   NaN      1      2  23.4500        S   Third   
889         1       1    male  26.0      0      0  30.0000        C   First   
890         0       3    male  32.0      0      0   7.7500        Q   Third   

       who  adult_male deck  embark_town alive  alone  
0      man        True  NaN  Southampton    no  False  
1    woman       False    C    Cherbourg   yes  False  
2    woman       False  NaN  Southampton   yes   True  
3    woman       False    C  Southampton   yes  False  
4      man        True  NaN  Southampton    no   True  
..     ...         ...  ...          ...   ...    ...  
886    man        True  NaN  Southampton    no   True  
887  woman       False    B  Southampton   yes   True  
888  woman       False  NaN  Southampton    no  False  
889    man        True    C    Cherbourg   yes   True  
890    man        True  NaN   Queenstown    no   True  

[891 rows x 15 columns]

> A base de dados possui 891 entradas e 15 colunas.

### 1.3 Dados faltantes

In [3]:
dados_faltantes = df.isnull().sum()
dados_faltantes

survived         0
pclass           0
sex              0
age            177
sibsp            0
parch            0
fare             0
embarked         2
class            0
who              0
adult_male       0
deck           688
embark_town      2
alive            0
alone            0
dtype: int64

> De acordo com os dados fornecidos, percebe-se que há 177 dados faltantes para a variável **age**, 2 para a variável **embarked** e 688 para a variável **deck**. As demais variáveis não possuem dados faltantes.

### 1.4 Dicionário de dados

In [5]:
dicionario = pd.read_csv("../data/external/dictionary.csv")
dicionario

variavel                                          descricao  \
0      survived          Indica se o passageiro sobreviveu ou não.   
1        pclass  Indica a classe do passageiro usando números i...   
2           sex                     Indica o gênero do passageiro.   
3           age                      Indica a idade do passageiro.   
4         sibsp  Indica a quantidade de irmãos e/ou conjuguês a...   
5         parch   indica a quantidade de pais e/ou filhos a bordo.   
6          fare     indica o valor da tarifa paga pelo passageiro.   
7      embarked        indica o porto de embarcação do passageiro.   
8         class       Indica a classe do passageiro usando string.   
9           who           Indica se é um homem, mulher ou criança.   
10   adult_male          Indica se o passageiro é um homem adulto.   
11         deck                         Indica o convés  do navio.   
12  embark_town          Indica a cidade de embarque d passageiro.   
13        alive                  Indica se o passageiro está vivo.   
14        alone             Indica se o passageiro estava sozinho.   

            tipo   subtipo  
0    qualitativa   nominal  
1    qualitativa   ordinal  
2    qualitativa   nominal  
3   quantitativa  contínua  
4   quantitativa  discreta  
5   quantitativa  discreta  
6   quantitativa  contínua  
7    qualitativa   nominal  
8    qualitativa   ordinal  
9    qualitativa   nominal  
10   qualitativa   nominal  
11  quanlitativa   ordinal  
12   qualitativa   nominal  
13   qualitativa   nominal  
14   qualitativa   nominal

> O dataset possui 15 variáveis, sendo 8 qualitativas nominais, 3 qualitativas ordinais, 2 quantitativas discretas e 2 quantitativas contínuas.

## 2. Perguntas de partida e hipóteses

## 3. Insights